connector-pagerduty/src/main/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyCloudEventDataExtractor.java

```diff
@@ -0,0 +1,53 @@
+package com.redhat.cloud.notifications.connector.pagerduty;
+
+import com.redhat.cloud.notifications.connector.CloudEventDataExtractor;
+import com.redhat.cloud.notifications.connector.authentication.AuthenticationDataExtractor;
+import io.vertx.core.json.JsonObject;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.camel.Exchange;
+import org.apache.commons.validator.routines.UrlValidator;
+
+import java.util.MissingResourceException;
+
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.TARGET_URL;
+import static org.apache.commons.validator.routines.UrlValidator.ALLOW_LOCAL_URLS;
+
+@ApplicationScoped
+public class PagerDutyCloudEventDataExtractor extends CloudEventDataExtractor {
+
+    public static final String AUTHENTICATION = "authentication";
+    public static final String URL = "url";
+
+    // HTTP URLs (or disabling SSL verification) must be permitted to run test cases
+    private static final UrlValidator URL_VALIDATOR = new UrlValidator(new String[]{"http", "https"}, ALLOW_LOCAL_URLS);
```

Comment:
If we're talking about unit tests, it should be possible to use `https` only. See this:
https://github.com/RedHatInsights/notifications-backend/blob/f313e719e473be0e79db9c93e210fee845d95224/connector-splunk/src/test/java/com/redhat/cloud/notifications/connector/splunk/SplunkConnectorRoutesTest.java#L100-L103
The tests from connector-pagerduty can be improved later, no need to change everything in this PR.

Reply:
It is possible to run it with only HTTPS, but it also requires setting `ALLOW_LOCAL_URLS`. Discussed it with @MikelAlejoBR  in [this thread](https://github.com/RedHatInsights/notifications-backend/pull/2869#discussion_r1711860897), I'm happy to configure it either way since HTTP won't be supported by PagerDuty anyways.